doc/usermanuals/DD4hep/chapters/basics.tex

```diff
@@ -743,52 +1118,165 @@ \subsection{Shape factories}
+\begin{minted}[frame=single,framesep=3pt,breaklines=true,tabsize=2,linenos,fontsize=\small]{c++}
+  <some_element type="Torus" r="value" rmin="value" rmax="value" startphi="value" deltaphi="value"/>
 \end{minted}
+Defaults are: \texttt{rmin=0},  \texttt{startphi=$\pi$},  \texttt{deltaphi=$2\times\pi$}.
+
 \item For a Sphere the constructor is:
 \begin{minted}[frame=single,framesep=3pt,breaklines=true,tabsize=2,linenos,fontsize=\small]{c++}
-Sphere(e.rmin(), e.rmax(), e.deltatheta(M_PI), e.phi(0e0),e.deltaphi(2.*M_PI))
+Sphere(e.rmin(), e.rmax(), e.starttheta(), e.deltatheta(), e.startphi(), e.deltaphi())
+\end{minted}
+The corrsponding XML snippet looks like this:
+\begin{minted}[frame=single,framesep=3pt,breaklines=true,tabsize=2,linenos,fontsize=\small]{c++}
+  <some_element type="Sphere" rmin="value" rmax="value"
+                starttheta="value" deltatheta="value" startphi="value" deltaphi="value"/>
 \end{minted}
+Defaults are: \texttt{rmin=0},  \texttt{starttheta=0},  \texttt{deltatheta=$\pi$},
+\texttt{startphi=0}, \texttt{deltaphi=$2\times\pi$}.
+
 \item For a Paraboloid the constructor is:
 \begin{minted}[frame=single,framesep=3pt,breaklines=true,tabsize=2,linenos,fontsize=\small]{c++}
-Paraboloid(e.rmin(0.0), e.rmax(), e.dz())
+Paraboloid(e.rmin(0), e.rmax(), e.dz())
+\end{minted}
+The corrsponding XML snippet looks like this:
+\begin{minted}[frame=single,framesep=3pt,breaklines=true,tabsize=2,linenos,fontsize=\small]{c++}
+  <some_element type="Paraboloid" rmin="value" rmax="value" dz="value"/>
+\end{minted}
+Defaults are: \texttt{rmin=0}.
+
+\item For a Hyperboloid the constructor is:
+\begin{minted}[frame=single,framesep=3pt,breaklines=true,tabsize=2,linenos,fontsize=\small]{c++}
+Hyperboloid(e.rmin(0), e.inner_stereo(), e.rmax(), e.outer_stereo, e.dz())
+\end{minted}
+The corrsponding XML snippet looks like this:
+\begin{minted}[frame=single,framesep=3pt,breaklines=true,tabsize=2,linenos,fontsize=\small]{c++}
+  <some_element type="Hyperboloid" rmin="value" inner_stereo="value" rmax="value" outer_stereo=="value" dz="value"/>
 \end{minted}
+
 \item For a PolyhedraRegular the constructor is:
 \begin{minted}[frame=single,framesep=3pt,breaklines=true,tabsize=2,linenos,fontsize=\small]{c++}
 PolyhedraRegular(e.numsides(), e.rmin(), e.rmax(), e.dz())
 \end{minted}
+The corrsponding XML snippet looks like this:
+\begin{minted}[frame=single,framesep=3pt,breaklines=true,tabsize=2,linenos,fontsize=\small]{c++}
+  <some_element type="PolyhedraRegular" numsides="value" rmin="value" rmax="value" dz="value"/>
+\end{minted}
+
+\item For a generic Polyhedra the constructor is:
+\begin{minted}[frame=single,framesep=3pt,breaklines=true,tabsize=2,linenos,fontsize=\small]{c++}
+PolyhedraRegular(e.numsides(), e.rmin(), e.startphi(), e.deltaphi(), vector<z>, vector<rmin>, vector<rmax>);
+\end{minted}
+The corrsponding XML snippet looks like this:
+\begin{minted}[frame=single,framesep=3pt,breaklines=true,tabsize=2,linenos,fontsize=\small]{c++}
```

Comment:
```suggestion
\begin{minted}[frame=single,framesep=3pt,breaklines=true,tabsize=2,linenos,fontsize=\small]{xml}
```

Reply:
You have in general many occasions where you have set highlighting of code snippets for XML to C++